src/test/java/myapp/HelperBuddy/DeadlineTest.java

```diff
@@ -0,0 +1,43 @@
+package myapp.helperbuddy;
+
+import org.junit.jupiter.api.Test;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+
+public class DeadlineTest {
+    @Test
+    public void testParsing() {
+        String taskData = "D | 0 | programming assignment | 02/09/2024 2359";
+        String[] parts = taskData.split(" \\| ");
+        String description = parts[2].trim();
+        LocalDateTime deadlineBy = null;
+        if (parts.length > 3) {
```

Comment:
Not sure how necessary this if statement is, since parts is defined by you and will always be the same?